Neos.Neos/Classes/Fusion/Helper/BackendHelper.php

```diff
@@ -40,6 +41,39 @@ public function interfaceLanguage(): string
         return $this->userService->getInterfaceLanguage();
     }
 
+    public function isEditMode(ActionRequest $request): bool
+    {
+        return ($request->getControllerPackageKey() === 'Neos.Neos'
+            && $request->getControllerName() === "Frontend\Node"
+            && $request->getControllerActionName() === 'edit'
+        );
+    }
+
+    public function isPreviewMode(ActionRequest $request): bool
+    {
+        return ($request->getControllerPackageKey() === 'Neos.Neos'
+            && $request->getControllerName() === "Frontend\Node"
+            && $request->getControllerActionName() === 'preview'
+        );
+    }
+
+    public function renderingModeCacheIdentifier(ActionRequest $request): string
+    {
+        if (
+            $request->getControllerPackageKey() === 'Neos.Neos'
+            && $request->getControllerName() === "Frontend\Node"
+            && ($request->getControllerActionName() === 'edit' || $request->getControllerActionName() === 'preview')
+        ) {
```

Comment:
```suggestion
        if ($this->isEditMode($request) || $this->isPreviewMode($request)) {
```

Reply:
This will check the controller and package twice ... not against it but also not a fan.

Reply:
I'm also in favor of preventing domain logic duplication.
`self::isEditMode(...) || self::isPreviewMode(...)` is much easier to comprehend and those string comparisons come with no remarkable overhead